ReviewChecklist.md

```diff
@@ -0,0 +1,153 @@
+# PR review checklist
+The Solidity compiler is a critical piece of infrastructure in the Ethereum ecosystem and for this
+reason our review process is quite strict.
+The list below is meant to reduce the workload on the core team by helping contributors self-identify
+common issues before they are inevitably pointed out in the review.
+It's also meant as a final checklist for reviewers to go through before approving a PR.
+
+## Before you submit a PR
+- [ ] **Is the issue ready to be worked on?**
+    - If the issue does not have a desirability label (`nice to have`, `should have`,
+        `must have eventually`, `must have`, `roadmap`) we still have not decided whether to implement it.
+    - If the issue `needs design`, we have not yet decided how it should be implemented.
+    - `good first issue candidate` means that the issue will potentially be a `good first issue`
+        eventually but at the moment it's not yet ready to be worked on.
+- [ ] **Is this a breaking change?** Breaking changes should be based on the `breaking` branch rather than on `develop`.
+- [ ] **Does the PR actually address the issue?**
+    - [ ] Mention the issue number in the PR description.
+        If the PR solves it completely, use the `Fixes #<issue number>` form so that Github will close the issue automatically
+    - [ ] Do not include issue number in the PR title, branch name or commit description.
+- [ ] When submitting a PR from a fork **create a branch and give it a good name.**
+    E.g. `fix-array-abi-encoding-bug`.
+    Avoid submitting PRs directly from the `develop` branch of a fork since it makes rebasing and fetching new changes harder.
```

Comment:
I think we should even go a step further and forbid this; i.e. if we see that an external contributor is trying to merge fork:develop into solidity:develop, to simply close it and request for changes to be in a separate branch. The phrasing here is kind of ambiguous - `Avoid submitting PRs...` - what does that even mean? Don't do it except in such and such cases, or is it just a nice way of saying 'don't do it'?

Reply:
I said "avoid", because technically you can do that. You just have to remember to then do `git push <fork> <your branch>:develop` when pushing and fetching changes to `develop` is a bit awkward. But you have a point. Someone who knows that, does not really need this advice. I made it stronger.

This first section is generally meant to be read before you submit the PR so you should just not do it :) Of course many people will only learn about it from us when we point them at the checklist and it's too late. I still wouldn't just close the PR. Maybe if the submitter has problems with it but usually those PRs are still salvageable and I merged them on many occasions. I don't like that you then just get `develop` and fork name in the merge commit description but that's just cosmetics.